tests/t0124_gateway_ipns_record_test.go

```diff
@@ -0,0 +1,115 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/ipfs/gateway-conformance/tooling/car"
+	. "github.com/ipfs/gateway-conformance/tooling/check"
+	"github.com/ipfs/gateway-conformance/tooling/ipns"
+	"github.com/ipfs/gateway-conformance/tooling/test"
+	. "github.com/ipfs/gateway-conformance/tooling/test"
+)
+
+func TestGatewayIPNSRecord(t *testing.T) {
+	fixture := car.MustOpenUnixfsCar("t0124/fixtures.car")
+	file := fixture.MustGetRoot()
+	fileCID := file.Cid()
+
+	// test_expect_success "Add the test directory & IPNS records" '
+	// ipfs dag import ../t0124-gateway-ipns-record/fixtures.car &&
+	// ipfs routing put /ipns/${IPNS_KEY} ../t0124-gateway-ipns-record/${IPNS_KEY}.ipns-record
+	// '
+	// TODO: what if you'd import with `t0124/simple-id.ipns-record`,
+	// then the loader "just" finds the file with `peer_id_simple-id.ipns-record`?
+	ipns := ipns.MustOpenIPNSRecordWithKey("t0124/k51qzi5uqu5dh71qgwangrt6r0nd4094i88nsady6qgd1dhjcyfsaqmpp143ab.ipns-record")
+	ipnsKey := ipns.Key()
+
+	tests := SugarTests{
+		// test_expect_success "Create and Publish IPNS Key" '
+		// curl "http://127.0.0.1:$GWAY_PORT/ipns/$IPNS_KEY" > curl_output_filename &&
+		// test_should_contain "Hello IPFS" curl_output_filename
+		// '
+		{
+			Name: "Create and Publish IPNS Key",
```

Comment:
I would rename this. We're neither creating, nor publishing an IPNS Key. Perhaps `Gateway Resolves IPNS Key` or something along those lines. We're just checking if the key is resolved to the right content.